packages/back-end/src/validators/experiments.ts

```diff
@@ -0,0 +1,183 @@
+  "dnf",
+  "won",
+  "lost",
+  "inconclusive",
+] as const;
+
+export type ExperimentResultsType = typeof experimentResultsType[number];
+
+export const experimentPhase = z
+  .object({
+    dateStarted: z.date(),
+    dateEnded: z.date().optional(),
+    name: z.string(),
+    reason: z.string(),
+    coverage: z.number(),
+    condition: z.string(),
+    savedGroups: z.array(savedGroupTargeting).optional(),
+    prerequisites: z.array(featurePrerequisite).optional(),
+    namespace: namespaceValue,
+    seed: z.string().optional(),
+    variationWeights: z.array(z.number()),
+  })
+  .strict();
+
+export type ExperimentPhase = z.infer<typeof experimentPhase>;
+
+export const experimentStatus = ["draft", "running", "stopped"] as const;
+
+export type ExperimentStatus = typeof experimentStatus[number];
+
+export const screenshot = z
+  .object({
+    path: z.string(),
+    width: z.number().optional(),
+    height: z.number().optional(),
+    description: z.string().optional(),
+  })
+  .strict();
+
+export type Screenshot = z.infer<typeof screenshot>;
+
+export const variation = z
+  .object({
+    id: z.string(),
+    name: z.string(),
+    description: z.string().optional(),
+    key: z.string(),
+    screenshots: z.array(screenshot),
+  })
+  .strict();
+
+export type Variation = z.infer<typeof variation>;
+
+export const attributionModel = [
+  "firstExposure",
+  "experimentDuration",
+] as const;
+
+export type AttributionModel = typeof attributionModel[number];
+
+export const implementationType = [
+  "visual",
+  "code",
+  "configuration",
+  "custom",
+] as const;
+
+export type ImplementationType = typeof implementationType[number];
+
+export const experimentNotification = [
+  "auto-update",
+  "multiple-exposures",
+  "srm",
+] as const;
+
+export type ExperimentNotification = typeof experimentNotification[number];
+
+export const metricOverride = z
+  .object({
+    id: z.string(),
+    windowType: windowTypeValidator.optional(),
+    windowHours: z.number().optional(),
+    delayHours: z.number().optional(),
+    winRisk: z.number().optional(),
+    loseRisk: z.number().optional(),
+    properPriorOverride: z.boolean().optional(),
+    properPriorEnabled: z.boolean().optional(),
+    properPriorMean: z.number().optional(),
+    properPriorStdDev: z.number().optional(),
+    regressionAdjustmentOverride: z.boolean().optional(),
+    regressionAdjustmentEnabled: z.boolean().optional(),
+    regressionAdjustmentDays: z.number().optional(),
+  })
+  .strict();
+
+export type MetricOverride = z.infer<typeof metricOverride>;
+
+export const experimentInterface = z
+  .object({
+    id: z.string(),
+    trackingKey: z.string(),
+    organization: z.string(),
+    project: z.string().optional(),
+    owner: z.string(),
+    datasource: z.string(),
+    exposureQueryId: z.string(),
+    /**
+     * @deprecated Always set to 'code'
+     */
+    implementation: z.enum(implementationType),
+    /**
+     * @deprecated
+     */
```

Comment:
The `@deprecated` comments when this was a Typescript interface were used by IDEs in code completion, but these are not.  I don't see a good alternative that works with Zod though, so I think it's ok, just a little unfortunate.